### PR TITLE
FOR REVIEW ONLY - Release v2.0.0

### PR DESCRIPTION
**Pull request by algobarb**
### Breaking changes

- Remove v1 algod API (`client.algod`) due to API end-of-life (2022-12-01). Instead, use v2 algod API (`client.v2.algod.algod`).
- Remove `cost` field in `DryrunTxnResult` in favor of 2 fields: `budget-added` and `budget-consumed`. `cost` can be derived by `budget-consumed - budget-added`.
- Remove logicsig templates (`logicTemplates`), `logic/langspec.json`, `logic.logic` depending on `langspec.json`.
- Regenerate algod models so every constructor requires an object to be passed in. Previously, only constructors with more than 4 argument specified this.
- Remove unused generated types: `CatchpointAbortResponse`, `CatchpointStartResponse`.
- Remove following methods in favor of the methods with `WithSuggestedParams` suffix:
  - `makePaymentTxn`, `makeKeyRegistrationTxn`, `makeAssetCreateTxn`, `makeAssetConfigTxn`, `makeAssetDestroyTxn`, `makeAssetFreezeTxn`, `makeAssetTransferTxn`.
- Remove `makeLogicSig` in favor of either using `LogicSigAccount` (preferred) or directly invoking `LogicSig` constructor.
- Remove `EncodedMultisigBlob` in favor of `EncodedSignedTransaction.

### Enhancements

- v2: Make breaking changes from v1 to v2.0.0 by @michaeldiamant in https://github.com/algorand/js-algorand-sdk/pull/717

**Full Changelog**: https://github.com/algorand/js-algorand-sdk/compare/v1.24.1...v2.0.0

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,25 @@
...
+- Regenerate algod models so every constructor requires an object to be passed in. Previously, only constructors with more than 4 argument specified this.
+- Remove unused generated types: `CatchpointAbortResponse`, `CatchpointStartResponse`.
+- Remove following methods in favor of the methods with `WithSuggestedParams` suffix:
+  - `makePaymentTxn`, `makeKeyRegistrationTxn`, `makeAssetCreateTxn`, `makeAssetConfigTxn`, `makeAssetDestroyTxn`, `makeAssetFreezeTxn`, `makeAssetTransferTxn`.
+- Remove `makeLogicSig` in favor of either using `LogicSigAccount` (preferred) or directly invoking `LogicSig` constructor.
+- Remove `EncodedMultisigBlob` in favor of `EncodedSignedTransaction.
+
+### Enhancements
+
+- v2: Make breaking changes from v1 to v2.0.0 by @michaeldiamant in https://github.com/algorand/js-algorand-sdk/pull/717
```
Comment (michaeldiamant) [inline CHANGELOG.md:19]: Optional - For this release, I think the already listed breaking changes account for the included changes.

```suggestion
```

File: README.md [hunk 2]
```diff
@@ -22,8 +22,8 @@ Include a minified browser bundle directly in your HTML like so:
```
Comment (michaeldiamant) [inline README.md:22]: Unrelated to the line:  As of https://github.com/algorand/js-algorand-sdk/pull/728/commits/e8a04c3c4cc6ef3e666d008bee539cf8c327e8d6, the PR shows _This branch is out-of-date with the base branch_.  Is it intentional or should the PR's branch be updated prior to merge?

I'll hold my approval until clarifying intended state.

File: CHANGELOG.md (hunk 1 above)
Comment (algobarb) [reply CHANGELOG.md]: removed this bit.

File: README.md (hunk 2 above)
Comment (algobarb) [reply README.md:22]: Yes it is intentional.

**Review by ahangsu — APPROVED** [review]

File: README.md (hunk 2 above)
Comment (michaeldiamant) [reply README.md:22]: Resolving - thanks for confirmation.

**Review by michaeldiamant — APPROVED** [review]

**Review by algochoi — APPROVED** [review]

**Pull request merged**